pyomo/gdp/plugins/multiple_bigm.py

```diff
@@ -636,8 +627,20 @@ def _calculate_missing_M_values(
                     if lower_M is None:
                         scratch.obj.expr = constraint.body - constraint.lower
                         scratch.obj.sense = minimize
-                        results = self._config.solver.solve(other_disjunct)
+                        results = self._config.solver.solve(
+                            other_disjunct, load_solutions=False
+                        )
                         if (
+                            results.solver.termination_condition
+                            is TerminationCondition.infeasible
+                        ):
```

Comment:
Are we sure that we want to check against "infeasible"?  For the "old" (current) solver interfaces, I am not sure if this is always a guarantee of infeasibility.  (In particular, I don't think that Ipopt will ever _guarantee_ infeasibility)

Reply:
I changed this so that we maintain a list of solvers we trust that 'infeasible' means 'proven infeasible' and we do this for those. For others (looking at you, ipopt), we just revert to the old behavior and scream about an unsuccessful solve.